File: fix_api_paths.php
<?php
$dir = new RecursiveDirectoryIterator('d:/01. Project/04. Website/pln/webapp/src');
$ite = new RecursiveIteratorIterator($dir);
$files = new RegexIterator($ite, '/^.+\.php$/i', RecursiveRegexIterator::GET_MATCH);

foreach ($files as $file) {
    $path = $file[0];
    $content = file_get_contents($path);
    
    // Replace fetch(`/api/...`) with fetch(`<?= API_URL ?>/api/...`)
    $replaced = preg_replace('/fetch\(\s*`\/api\//', 'fetch(`<?= API_URL ?>/api/', $content);
    
    // Replace `/api/...` string literals inside JS (usually in `let url = ...`)
    $replaced = preg_replace('/`\/api\//', '`<?= API_URL ?>/api/', $replaced);
    
    if ($replaced !== $content) {
        file_put_contents($path, $replaced);
        echo "Fixed hardcoded /api/ in: $path\n";
    }
}
?>

File: fix_api_paths.py
import os
import re

dir_path = r'd:\01. Project\04. Website\pln\webapp\src'
pattern = re.compile(r'/api/')

for root, dirs, files in os.walk(dir_path):
    for file in files:
        if file.endswith('.php'):
            filepath = os.path.join(root, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()
            
            replaced = pattern.sub(r'<?= API_URL ?>/api/', content)
            
            if replaced != content:
                with open(filepath, 'w', encoding='utf-8') as f:
                    f.write(replaced)
                print(f'Fixed {filepath}')
